middleware/superkey_destroy.go

```diff
@@ -0,0 +1,87 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/RedHatInsights/sources-api-go/dao"
+	"github.com/RedHatInsights/sources-api-go/jobs"
+	"github.com/labstack/echo/v4"
+)
+
+/*
+	This middleware intercepts a superkey-related source on its way through the
+	stack and handles whether the requested resource is superkey related.
+
+	If it is then we will queue up a job that sends the request over to the
+	worker (to delete the resources in amazon), wait 15 seconds, then destroy
+	the actual resources.
+*/
+func SuperKeyDestroySource(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		tenantId, ok := c.Get("tenantID").(int64)
+		if !ok {
+			return fmt.Errorf("failed to pull tenant from request")
+		}
+
+		id, err := strconv.ParseInt(c.Param("id"), 10, 64)
+		if err != nil {
+			return err
+		}
+
+		s := dao.GetSourceDao(&id)
```

Comment:
Classic "bug introduced by rebase" behavior here :facepalm: I totally did this on accident the first time, then rebased bringing in the DAO changes, and re-introduced the bug. 

The good news is that it would have shown up really quick lol